diff --git a/gopro_overlay/widgets/video.py b/gopro_overlay/widgets/video.py
index e9649ce..35b486d 100755
--- a/gopro_overlay/widgets/video.py
+++ b/gopro_overlay/widgets/video.py
@@ -42,15 +42,13 @@ class VideoFrameSource:
         if seconds < 0:
             return None

-        wanted_index = int(seconds * self.fps)
+        wanted_index = round(seconds * self.fps)

         if self.process is None or self.next_index is None or wanted_index < self.next_index:
-            self._open(seconds)
-            wanted_index = 0
+            self._open(seconds, wanted_index)

         if wanted_index > self.next_index + int(self.fps * 2):
-            self._open(seconds)
-            wanted_index = 0
+            self._open(seconds, wanted_index)

         while self.next_index < wanted_index:
             if self._read_frame_bytes() is None:
@@ -64,7 +62,7 @@ class VideoFrameSource:
         self.next_index += 1
         return Image.frombytes(mode="RGBA", size=self.dimensions.tuple(), data=data)

-    def _open(self, seconds: float):
+    def _open(self, seconds: float, next_index: int):
         self.close()
         cmd = [
             str(self.ffmpeg._path()),
@@ -76,7 +74,7 @@ class VideoFrameSource:
             "-",
         ]
         self.process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
-        self.next_index = 0
+        self.next_index = next_index

     def _read_frame_bytes(self):
         data = self.process.stdout.read(self.frame_bytes)
